components/pluralrules/src/operands.rs

```diff
@@ -0,0 +1,275 @@
+            n: absolute_value,
+            i: integer_digits,
+            v: num_fraction_digits0,
+            w: num_fraction_digits,
+            f: fraction_digits0,
+            t: fraction_digits,
+        })
+    }
+}
+
+macro_rules! impl_integer_type {
+    ($ty:ident) => {
+        impl From<$ty> for PluralOperands {
+            fn from(input: $ty) -> Self {
+                PluralOperands {
+                    n: input as f64,
+                    i: input as u64,
+                    v: 0,
+                    w: 0,
+                    f: 0,
+                    t: 0,
+                }
+            }
+        }
+    };
+    ($($ty:ident)+) => {
+        $(impl_integer_type!($ty);)+
+    };
+}
+
+macro_rules! impl_signed_integer_type {
+    ($ty:ident) => {
+        impl TryFrom<$ty> for PluralOperands {
+            type Error = OperandsError;
+            fn try_from(input: $ty) -> Result<Self, Self::Error> {
+                let x = input.checked_abs().ok_or(OperandsError::Invalid)?;
+                Ok(PluralOperands {
+                    n: x as f64,
+                    i: x as u64,
+                    v: 0,
+                    w: 0,
+                    f: 0,
+                    t: 0,
+                })
+            }
+        }
+    };
+    ($($ty:ident)+) => {
+        $(impl_signed_integer_type!($ty);)+
+    };
+}
+
+// This algorithms allows us to use the Rust's decimal to float parser
+// using on-stack buffer.
+//
+// Once we start relying on Ryu, we should consider swapping it to use that.
+fn get_dec_str_from_f64(absolute_value: f64, buffer: &mut [u8]) -> Result<&str, IOError> {
+    let mut slice = &mut buffer[..];
+    write!(slice, "{}", absolute_value)?;
+    let remaining_len = slice.len();
+    let written = buffer.len() - remaining_len;
+
+    let abs_str = &buffer[0..written];
+
+    let sep_idx = abs_str
+        .iter()
+        .position(|b| b == &b'.')
+        .expect("Failed to find a separator in f64.");
+
+    // # Safety
+    //
+    // We are dealing with ASCII digits.
+    Ok(unsafe { std::str::from_utf8_unchecked(&abs_str[(sep_idx + 1)..]) })
+}
+
+macro_rules! impl_float_type {
+    ($ty:ident) => {
+        impl TryFrom<$ty> for PluralOperands {
+            type Error = OperandsError;
+            fn try_from(input: $ty) -> Result<Self, Self::Error> {
+                let absolute_value = input.abs() as f64;
+
+                let (
+                    integer_digits,
+                    num_fraction_digits0,
+                    num_fraction_digits,
+                    fraction_digits0,
+                    fraction_digits,
+                ) = if (absolute_value.round() - absolute_value).abs() > f64::EPSILON {
+                    let mut buffer = [0u8; 32];
+                    let dec_str = get_dec_str_from_f64(absolute_value, &mut buffer)?;
+                    let num_fraction_digits = dec_str.len();
+
+                    let fraction_digits = u64::from_str(&dec_str)?;
```

Comment:
So you are converting back and forth to a string in order to split the f64 into integer and fraction parts?  That's the same fundamental operations that FixedDecimal will be doing.  Therefore, there is no *technical* advantage to keeping the f64 code path.

Reply:
Sure, therefore I hope to land this, then switch to `FixedDecimal`, benchmark and remove this.